File: src/HookBridge.Dashboard/src/types/failedEvent.ts
import type { PagedRequest } from './pagination';

export type FailedEventStatus = 'DLQ' | 'RetryRequested';

export type FailedEventResponse = {
  id: string;
  tenantId: string;
  eventId: string;
  subscriptionId: string;
  eventType: string;
  targetUrl: string;
  reason: string;
  finalAttemptNumber: number;
  lastHttpStatusCode: number | null;
  lastErrorMessage: string | null;
  status: FailedEventStatus | string;
  failedAt: string;
  correlationId: string | null;
  createdAt: string;
  updatedAt: string | null;
};

export type FailedEventSearchRequest = PagedRequest & {
  eventId?: string;
  subscriptionId?: string;
  eventType?: string;
  status?: FailedEventStatus | '';
  fromDate?: string;
  toDate?: string;
};

File: src/HookBridge.Domain/Entities/FailedEvent.cs
namespace HookBridge.Domain.Entities;

/// <summary>
/// Represents an event that permanently failed delivery and was moved to the dead letter queue.
/// </summary>
public sealed class FailedEvent : BaseEntity
{
    public string TenantId { get; set; } = string.Empty;

    public string EventId { get; set; } = string.Empty;

    public string SubscriptionId { get; set; } = string.Empty;

    public string EventType { get; set; } = string.Empty;

    public string TargetUrl { get; set; } = string.Empty;

    public string Reason { get; set; } = string.Empty;

    public int FinalAttemptNumber { get; set; }

    public int? LastHttpStatusCode { get; set; }

    public string? LastErrorMessage { get; set; }

    public string Status { get; set; } = string.Empty;

    public DateTime FailedAt { get; set; }

    public string? CorrelationId { get; set; }
}
